spec/framework/config/fact_groups_spec.rb

```diff
@@ -54,6 +54,20 @@
       end
     end
 
+    context 'with blocked `legacy` group' do
+      before do
+        stub_const('Facter::Config::FACT_GROUPS', 'legacy' => %w[legacy_fact1 legacy_fact2])
+        allow(config_reader).to receive(:block_list).and_return(%w[legacy])
+      end
+
+      # legacy facts are excluded because they are blocked by another mechanism that takes into account the fact's type
+      it 'excludes legacy facts' do
+        blk_list = fact_groups.new
+
+        expect(blk_list.blocked_facts).to be_empty
+      end
```

Comment:
is this test working properly? I removed the stubs in the `before` block and it is still passing.

Reply:
The stubs mock facter's default groups and the config from `facter.conf`. `blocked_facts` method takes the facts and groups from `facter.conf` and tries to break the groups in facts (using FACT_GROUPS from [config.rb](https://github.com/puppetlabs/facter/blob/5f00e96aaff19bfa5af85d195624bb873463233f/lib/facter/config.rb#L44), which is also mocked) 

The new condition just tells that `blocked_facts` should not break the `legacy` group in facts, it should just ignore it.

The stubs tell what group to transform into facts (in this case `legacy` group) and how to do this (FACT_GROUPS defines what facts belong to a group). When you removed the mocks you basically removed `facter.conf` and the `blocklist: ['legacy']` condition. In this case, the new logic does nothing.